### ISSUE TITLE
Support for Non-HTTP Triggers

### ISSUE THREAD

**Issue by calebschoepp** [spinframework/spin-operator#15]
One thing that would be incredibly value in a kubernetes deployment is the ability to use alternative triggers - this is currently blocked on https://github.com/deislabs/containerd-wasm-shims/issues/193 - but we should run through a hypothetical example to understand if we're over-optimizing our api surface towards http.

**Comment by calebschoepp** [comment]
I don't think we can close this yet. AFAIK we don't support arbitrary triggers in the shim which is what this is tracking.

If we no longer care about that or want to track that in the shim repo then those would be good reasons to close this issue.

**Comment by endocrimes** [comment]
We have a bit of work to do in the operator to potentially configure supported triggers (the shim automatically picking things up from the artifact rocks) but do we want to reason through things like "custom http ports"? Or health checks for non http triggers? etc
